test/unit/mongo_client.test.js

```diff
@@ -976,35 +1019,162 @@ describe('MongoOptions', function () {
-          process.env['MONGODB_LOG_MAX_DOCUMENT_LENGTH'] = '155';
-          const client = new MongoClient('mongodb://a/', {
-            [loggerFeatureFlag]: true,
-            mongodbLogMaxDocumentLength: 290
+    });
+    describe('mongodbLogMaxDocumentLength', function () {
+      context('when mongodbLogMaxDocumentLength is in options', function () {
+        context('when env option for MONGODB_LOG_MAX_DOCUMENT_LENGTH is not provided', function () {
+          it('should store value for maxDocumentLength correctly', function () {
+            const client = new MongoClient('mongodb://a/', {
+              [loggerFeatureFlag]: true,
+              mongodbLogMaxDocumentLength: 290
+            });
+            expect(client.options.mongoLoggerOptions.maxDocumentLength).to.equal(290);
+          });
+          it('should throw error for negative input', function () {
+            expect(
+              () =>
+                new MongoClient('mongodb://a/', {
+                  [loggerFeatureFlag]: true,
+                  mongodbLogMaxDocumentLength: -290
+                })
+            ).to.throw(MongoParseError);
+          });
+        });
+        context('when env option for MONGODB_LOG_MAX_DOCUMENT_LENGTH is provided', function () {
+          beforeEach(function () {
+            process.env['MONGODB_LOG_MAX_DOCUMENT_LENGTH'] = '155';
+          });
+
+          afterEach(function () {
+            process.env['MONGODB_LOG_MAX_DOCUMENT_LENGTH'] = undefined;
+          });
+
+          it('should store value for maxDocumentLength correctly (client option value takes precedence)', function () {
+            const client = new MongoClient('mongodb://a/', {
+              [loggerFeatureFlag]: true,
+              mongodbLogMaxDocumentLength: 290
+            });
+            expect(client.options.mongoLoggerOptions.maxDocumentLength).to.equal(290);
+          });
+          it('should throw error for negative input', function () {
+            expect(
+              () =>
+                new MongoClient('mongodb://a/', {
+                  [loggerFeatureFlag]: true,
+                  mongodbLogMaxDocumentLength: -290
+                })
+            ).to.throw(MongoParseError);
+          });
+        });
+      });
+      context('when mongodbLogMaxDocumentLength is not in options', function () {
+        context('when env option for MONGODB_LOG_MAX_DOCUMENT_LENGTH is not provided', function () {
+          it('should store value for default maxDocumentLength correctly', function () {
+            const client = new MongoClient('mongodb://a/', {
+              [loggerFeatureFlag]: true
+            });
+            expect(client.options.mongoLoggerOptions.maxDocumentLength).to.equal(1000);
+          });
+        });
+        context('when env option for MONGODB_LOG_MAX_DOCUMENT_LENGTH is provided', function () {
+          afterEach(function () {
+            process.env['MONGODB_LOG_MAX_DOCUMENT_LENGTH'] = undefined;
```

Comment:
```suggestion
            delete process.env.MONGODB_LOG_MAX_DOCUMENT_LENGTH;
```

Any value that is set on process.env is stringified, so you'll get "undefined" as a value of MONGODB_LOG_MAX_DOCUMENT_LENGTH. I believe what you want here is to remove the setting altogether. 

Also, you don't have to use computed key syntax here, dot access works too.

Reply:
Sounds good, I've changed all of them to dot access since they aren't computed, and also used the `delete` keyword instead of setting to `'undefined'`